File: Python Codes/ISBN code verification.py
def isValidISBN(isbn):
	if len(isbn) != 10:
		return False
	
	sum = 0
	for i in range(9):
		sum += int(isbn[i]) * (10 - i)
	
	if(isbn[9] == 'X'):
		sum+=10
	else:
		sum+=int(isbn[9])*10
	
	return (sum % 11 == 0)

code = input("Enter the code to be verified: ")
if isValidISBN(code):
	print('Valid')
else:
	print("Invalid")

File: Cpp Codes/ISBN code verification.cpp
#include <iostream>
using namespace std;

bool isValidISBN(string isbn)
{
    if (isbn.length() != 10)
        return false;

    int sum = 0;
    for (int i = 0; i < 9; i++)
    {
        int digit = isbn[i] - '0';
        if (digit < 0 || digit > 9)
            return false;
        sum += (digit * (10 - i));
    }

    char end = isbn[9];
    if (end != 'X' && (end < '0' || end > '9'))
        return false;

    if (end == 'X')
        sum += 10;
    else
        sum += (end - '0');

    return (sum % 11 == 0);
}

int main()
{
    string isbn;
    cout << "Enter ISBN Number:";
    cin >> isbn;

    if (isValidISBN(isbn))
        cout << "Valid";
    else
        cout << "Invalid";
    return 0;
}